types/react-native/index.d.ts

```diff
@@ -3410,6 +3410,166 @@ export interface ImageStatic extends NativeMethodsMixin, React.ComponentClass<Im
 }
 
 /**
+ * @see https://facebook.github.io/react-native/docs/flatlist.html#props
+ */
+export interface FlatListProperties extends React.Props<FlatListStatic> {
+
+    /**
+     * Rendered in between each item, but not at the top or bottom
+     */
+    ItemSeparatorComponent?: React.ComponentClass<any>
+
+    /**
+     * Rendered at the bottom of all the items.
+     */
+    ListFooterComponent?: React.ComponentClass<any>
+
+    /**
+     * Rendered at the top of all the items.
+     */
+    ListHeaderComponent?: React.ComponentClass<any>
+
+    /**
+     * Optional custom style for multi-item rows generated when numColumns > 1
+     */
+    columnWrapperStyle?: ViewStyle
+
+    /**
+     * For simplicity, data is just a plain array. If you want to use something else,
+     * like an immutable list, use the underlying VirtualizedList directly.
+     */
+    data?: any[]
```

Comment:
This isn’t optional, but it is nullable: https://github.com/facebook/react-native/blob/v0.43.1/Libraries/CustomComponents/Lists/FlatList.js#L66